### PR TITLE
CI: Notify failures in separate pipeline

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Changes slack notifications to execute in a separate pipeline, triggered by failure and depending on the other master  pipelines.

### CODE REVIEW

File: scripts/lib.star [hunk 1]
```diff
@@ -145,6 +145,12 @@ def master_pipelines(edition):
             name='publish-master', edition=edition, trigger=trigger, steps=publish_steps,
             depends_on=['build-master', 'windows-master',], install_deps=False, version_mode=version_mode,
         ))
+
+        notifyTrigger = dict(trigger, status = ['failure'])
```
Comment (aknuds1) [inline scripts/lib.star:149]: Can you write in snake case instead of camel case, since it's standard in Starlark?

**Review by aknuds1 — APPROVED** [review]
Looks good I think, but Starlark should have snake case, just like Python.

File: scripts/lib.star (hunk 1 above)
Comment (marefr) [reply scripts/lib.star:149]: Sure will fix. I thought you could do `trigger + { 'status': ['failure']` similar to jsonnet, but it didn't work so used dict instead. Do you know?

**Comment by marefr** [comment]
@aknuds1 does it make sense to have trigger of pipeline by branch, by event and by status?

**Comment by aknuds1** [comment]
@marefr have to look into it after eating dinner.

**Comment by marefr** [comment]
@aknuds1 no worries. I'm off for dinner as well. Pushed some enterprise/downstream fixes as well

File: scripts/lib.star [hunk 2]
```diff
@@ -204,6 +216,22 @@ def pipeline(
 
     return pipeline
 
+def notifyPipeline(name, slack_channel, trigger, depends_on=[]):
```
Comment (aknuds1) [inline scripts/lib.star:219]:
```suggestion
def notify_pipeline(name, slack_channel, trigger, depends_on=[]):
```

File: scripts/lib.star [hunk 3]
```diff
@@ -145,6 +145,12 @@ def master_pipelines(edition):
             name='publish-master', edition=edition, trigger=trigger, steps=publish_steps,
             depends_on=['build-master', 'windows-master',], install_deps=False, version_mode=version_mode,
         ))
+
+        notify_trigger = dict(trigger, status = ['failure'])
+        pipelines.append(notifyPipeline(
```
Comment (aknuds1) [inline scripts/lib.star:150]:
```suggestion
        pipelines.append(notify_pipeline(
```

File: scripts/lib.star [hunk 4]
```diff
@@ -165,6 +171,12 @@ def master_pipelines(edition):
             version_mode=version_mode,
         ))
 
+        notify_trigger = dict(trigger, status = ['failure'])
+        pipelines.append(notifyPipeline(
```
Comment (aknuds1) [inline scripts/lib.star:175]:
```suggestion
        pipelines.append(notify_pipeline(
```

**Review by aknuds1 — APPROVED** [review]
LGTM

File: scripts/lib.star (hunk 1 above)
Comment (aknuds1) [reply scripts/lib.star]: @marefr `dict` is probably fine. There's also a dict literal merging syntax in Python, but not sure if Starlark supports it.

**Comment by aknuds1** [comment]
> @aknuds1 does it make sense to have trigger of pipeline by branch, by event and by status?

I think the original trigger by branch/push makes sense at least? Unless I'm mistaken, it means to listen for pushes to master.

**Pull request merged**